Index: unzip.c
===================================================================
--- trunk/libbb/unzip.c	(revision 215)
+++ trunk/libbb/unzip.c	(working copy)
@@ -1012,7 +1012,8 @@
 extern void gz_close(int gunzip_pid)
 {
 	if (kill(gunzip_pid, SIGTERM) == -1) {
-		error_msg_and_die("***  Couldnt kill old gunzip process *** aborting");
+		perror("Couldnt kill gzip process (already dead?)");
+		/* DO NOT FAIL, DUH */
 	}
 
 	if (waitpid(gunzip_pid, NULL, 0) == -1) {
